js/src/Components/Widgets/LinkWidget.js

```diff
@@ -96,9 +95,13 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
 
       if (parsedUri.pathname === '/') {
         displayString = '<front>' + uri.substr(uri.indexOf('/') + 1);
+      } else if (parsedUri.pathname.indexOf('/<front>') >= 0) {
+        // If the user inputs `<front>#some-anchor`, the saved data on the entity is saved as
+        // `internal:/<front>#some-anchor` and the pathname is `/<front>#some-anchor`. We want
+        // to clean that up for users.
```

Comment:
TIL. Thanks for the comment.

Reply:
I'm not 100% sure if that how regular fields handle the data, but its how the data is saved at this time. so :man_shrugging: